modules/framework/galasa-parent/dev.galasa.framework.resource.management/src/main/java/dev/galasa/framework/resource/management/internal/ResourceManagement.java

```diff
@@ -161,6 +196,102 @@ public void run(Properties bootstrapProperties, Properties overrideProperties) t
         }
     }
 
+    // Package-level to allow unit testing
+    void loadMonitorBundles(IBundleManager bundleManager, String stream, IStreamsService streamsService) throws FrameworkException {
+        if (stream != null && !stream.isBlank()) {
+            loadRepositoriesFromStream(stream.trim(), streamsService);
+        }
+
+        Set<String> bundlesToLoad = getResourceMonitorBundles();
+
+        // Load the resulting bundles that have the IResourceManagementProvider service
+        for (String bundle : bundlesToLoad) {
+            if (!bundleManager.isBundleActive(bundleContext, bundle)) {
+                logger.info("ResourceManagement - loading bundle: " + bundle);
+
+                bundleManager.loadBundle(repositoryAdmin, bundleContext, bundle);
+
+                logger.info("ResourceManagement - bundle '" + bundle + "' loaded OK");
+            }
+        }
+    }
+
+    private void loadRepositoriesFromStream(String streamName, IStreamsService streamsService) throws FrameworkException {
+        IStream stream = streamsService.getStreamByName(streamName);
+
+        // Add the stream's OBR to the repository admin
+        String commaSeparatedObrs = stream.getObrLocation();
+        if (commaSeparatedObrs == null || commaSeparatedObrs.isBlank()) {
+            throw new FrameworkException("No OBR has been configured into the provided test stream");
+        }
+
+        for (String obr : commaSeparatedObrs.split(",")) {
```

Comment:
I would prefer to see a method which gets an array of OBRs, separate from a loop which adds those to the repository admin.

Comma-separator list handling seems noise in this method when it could be moved to  a dedicated method devoted to get the list of OBRs from the stream.

Why can't the stream object IStream return a list ? seems strange that the function isn't pushed into there.

Also getObrLocation() is a weird name if it can return a comma separated list. Re-name to something more appropriate, or do as suggested above ?

Reply:
Sounds good - updated the IStream object to return a List of OBRs and its setter handles the splitting of OBRs in https://github.com/galasa-dev/galasa/pull/236/commits/d7bf0fdb58a20eeca9b4d071476e26ee6797180d